### ISSUE TITLE
Awaitable instances not correctly detected

### ISSUE THREAD

**Issue by petri** [microsoft/pylance-release#431]
## Environment data

MacOS Catalina 10.15.6, Pylance v2020.9.6

## Expected behaviour

Do not flag the awaitable class instance as not awaitable

## Actual behaviour

Awaitable class instance is marked as non-awaitable.

## Logs

![image](https://user-images.githubusercontent.com/64672/94452275-07b75e00-01b8-11eb-9ddb-733c1f652af6.png)

## Code Snippet / Additional information

```python
from collections.abc import Awaitable


class MyAwaitable(Awaitable):
    "test awaitable"

    def __await__(self):
        yield


async def test():
    "test coro"

    await MyAwaitable()
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of Pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.7, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202097-30-september-2020
